### PR TITLE
feat(fulfillment): shipping options context field

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**

### CODE REVIEW

File: packages/fulfillment/src/services/fulfillment-module-service.ts [hunk 1]
```diff
@@ -137,6 +137,49 @@ export default class FulfillmentModuleService<
...
+      filters.context = optionsContext
+
+      return await this.listShippingOptionsForContext(
+        filters,
+        config,
+        sharedContext
+      )
+    }
+
+    return await super.listShippingOptions(filters, config, sharedContext)
```
Comment (adrien2p) [inline packages/fulfillment/src/services/fulfillment-module-service.ts:182]: q: if the `shipping_options_context` is not present, but the context is present in the filters, would it break the list here?

File: packages/fulfillment/src/services/fulfillment-module-service.ts [hunk 2]
```diff
@@ -137,6 +137,49 @@ export default class FulfillmentModuleService<
...
+      return
+    }
+
+    const shippingOptionsContext = filters.context ?? {}
+
+    // cleanup virtual field "shipping_options_context"
+    config.relations?.splice(fieldIdx, 1)
+    delete filters.context
+
+    return shippingOptionsContext
```
Comment (adrien2p) [inline packages/fulfillment/src/services/fulfillment-module-service.ts]:
```suggestion
    const fieldIdx = config.relations?.indexOf("shipping_options_context")
    const shouldListWithContext = fieldIdx > -1
    const shippingOptionsContext = filters.context ?? {}

    delete filters.context

    if (!shouldListWithContext) {
      return
    }

    // cleanup virtual field "shipping_options_context"
    config.relations?.splice(fieldIdx, 1)

    return shippingOptionsContext
```

Comment (adrien2p) [reply packages/fulfillment/src/services/fulfillment-module-service.ts]: I moved the delete above just to be sure that we do not end up calling the simple list with a context if the virtual relation is not present, let me know wyt

**Review by riqwan — APPROVED** [review]
nice!

**Pull request merged**